src/resolve/resolved-mdns.c

```diff
@@ -262,12 +256,19 @@ int manager_mdns_ipv4_fd(Manager *m) {
                 goto fail;
         }
 
+        /* try to bind without SO_REUSEADDR option to detect another mDNS responder */
         r = bind(m->mdns_ipv4_fd, &sa.sa, sizeof(sa.in));
         if (r < 0) {
                 r = -errno;
                 goto fail;
         }
 
+        r = setsockopt(m->mdns_ipv4_fd, SOL_SOCKET, SO_REUSEADDR, &one, sizeof(one));
```

Comment:
> if systemd-resolved crashes with some mdns TCP transfers outstanding, or if the remote crashes, and we try to restart systemd-resolved, it would be able to bind to the socket.

Right. My current PR does not support these cases. However in [RFC6762 Section 15.1](https://tools.ietf.org/html/rfc6762#section-15.1),

> One way to detect if there is another Multicast DNS implementation already running is to attempt binding to port 5353 without using SO_REUSEPORT and/or SO_REUSEADDR, and if that fails it indicates that some other socket is already bound to this port.

So, another choice is that we first try to `bind` without `SO_REUSEADDR` and if it fails, show warning and retry to `bind` with `SO_REUSEADDR`. WDYT? The similar logic is also used by avahi. See [avahi-core/socket.c](https://github.com/lathiat/avahi/blob/2b0972d47533803c62f7d2b4d359840f55cd58dd/avahi-core/socket.c#L188)